mlflow/models/evaluation/base.py

```diff
@@ -75,7 +75,8 @@ class EvaluationMetric:
             def eval_fn(
                 eval_df: Union[pandas.Dataframe, pyspark.sql.DataFrame],
                 builtin_metrics: Dict[str, float],
-            ) -> float:
+                metrics: Dict[str, MetricValue],
```

Comment:
I don't think this is quite what we want. In order to use the new `metrics`, users will have to define this placeholder `builtin_metrics` that they'll never use. Can we require the user to specify a type hint if they want to use `Dict[str, MetricValue]` instead?

See https://github.com/mlflow/mlflow/pull/9439#discussion_r1310759860